### ISSUE TITLE
launch-darkly c-server-sdk compilation

### ISSUE THREAD

**Issue by vsubram95** [launchdarkly/c-server-sdk#59]
**Description**
Compilation of launchdarkly c-server-sdk fails on using the command
`cmake --build .`

**Steps To reproduce**
execute the below commands within the DockerFile(docker desktop: v4.10.1) installed in a mac with m1
```
#!/usr/bin/env bash
set -ev                                                                                                                                                                                                           
git clone https://github.com/launchdarkly/c-server-sdk.git
git clone https://github.com/launchdarkly/c-server-sdk.git
cd c-server-sdk
mkdir build && cd build
cmake -D BUILD_TESTING=OFF -D BUILD_SHARED_LIBS=ON ..
cmake --build .
```
**Error**
```
#17 9.848 [ 62%] Building C object CMakeFiles/ldserverapi.dir/src/integrations/file_data.c.o
#17 9.864 /tmp/ci-LmJeFjvSre/c-server-sdk/src/integrations/file_data.c: In function 'readFile':
#17 9.864 /tmp/ci-LmJeFjvSre/c-server-sdk/src/integrations/file_data.c:74:15: error: comparison is always false due to limited range of data type [-Werror=type-limits]
#17 9.864    74 |         if (c == EOF) {
#17 9.864       |               ^~
#17 9.873 cc1: all warnings being treated as errors
#17 9.875 make[2]: *** [CMakeFiles/ldserverapi.dir/build.make:271: CMakeFiles/ldserverapi.dir/src/integrations/file_data.c.o] Error 1
#17 9.875 make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/ldserverapi.dir/all] Error 2
#17 9.876 make: *** [Makefile:130: all] Error 2
```
**SDK version**
Any c-server-sdk on and after version 2.6.0

**Comment by cwaldren-ld** [comment]
Hi @vsubram95 , I'm not sure what Dockerfile you are referring to - can you send a link, or copy it here?

_Filed internally as 161801._

**Comment by vsubram95** [comment]
Hey @cwaldren-ld, Thanks for your response. What i means is if we try creating the docker image with launchdarkly baked into it, we would need a DockerFile like the below
```
FROM openresty/openresty:1.19.9.1-10-focal as ld_builder
RUN apt-get update && apt-get install -y \
    luajit libluajit-5.1-dev lua-ldoc zip \
    ca-certificates curl libpcre3 libcurl4-openssl-dev \
    cmake libhiredis-dev git build-essential libpcre3-dev
COPY ld/install-sdk.sh .
```

and install-sdk.sh is,
```
#!/usr/bin/env bash
set -ev                                                                                                                                                                                                           
git clone https://github.com/launchdarkly/c-server-sdk.git
git clone https://github.com/launchdarkly/c-server-sdk.git
cd c-server-sdk
mkdir build && cd build
cmake -D BUILD_TESTING=OFF -D BUILD_SHARED_LIBS=ON ..
cmake --build .
```
In the above configuration, I am getting the error i mentioned above.

**Comment by cwaldren-ld** [comment]
@vsubram95 

You are using `openresty/openresty:1.19.9.1-10-focal`, which contains GCC `v9.4.0`.

I tried using older `openresty/openresty:1.17.8.2-5-focal`, where GCC version is `v9.3.0`. There is **no error** in this version. 

I will release an SDK update this week to fix it.

If you cannot wait for new release, you could try:
- Use older version of openresty
- Patch the source code on line 74 of `c-server-sdk/src/integrations/file_data.c`; change `char c;` to `int c;`

**Comment by vsubram95** [comment]
Thanks @cwaldren-ld. Really appreciate it.

**Comment by cwaldren-ld** [comment]
Hi @vsubram95 , this has been fixed in [2.8.1](https://github.com/launchdarkly/c-server-sdk/releases/tag/2.8.1).

**Comment by vsubram95** [comment]
Thanks @cwaldren-ld. I can confirm that that it works.
